src/com/foamdev/demos/bootstrap/ControllerForms.js

```diff
@@ -0,0 +1,194 @@
+            start('input').attrs({ type: 'text' , id: 'validationCustom04' , placeholder: 'State' , required: 'true' }).addClass('form-control').end().
+            start('div').addClass('invalid-feedback').add('Please provide a valid State.').end().
+          end().
+          /*
+          <div class="col-md-3 mb-3">
+            <label for="validationCustom05">Zip</label>
+            <input type="text" class="form-control" id="validationCustom05" placeholder="Zip" required>
+            <div class="invalid-feedback">
+              Please provide a valid zip.
+            </div>
+          </div>
+        </div>*/
+          start('div').addClass('col-md-3').addClass('mb-3').
+            start('label').attrs({ for: 'validationCustom05' }).add('Zip').end().
+            start('input').attrs({ type: 'text' , id: 'validationCustom05' , placeholder: 'Zip' , required: 'true' }).addClass('form-control').end().
+            start('div').addClass('invalid-feedback').add('Please provide a valid Zip.').end().
+          end().
+        end().
+        /*
+        <div class="form-group">
+          <div class="form-check">
+            <input class="form-check-input" type="checkbox" value="" id="invalidCheck" required>
+            <label class="form-check-label" for="invalidCheck">
+              Agree to terms and conditions
+            </label>
+            <div class="invalid-feedback">
+              You must agree before submitting.
+            </div>
+          </div>
+        </div>
+        <button class="btn btn-primary" type="submit">Submit form</button>
+      </form>*/
+
+        start('div').addClass('form-group').
+          start('div').addClass('form-check').
+            start('input').attrs({ type: 'checkbox', id: 'invalidCheck', value: '', required: 'true' }).addClass('form-check-input').end().
+            start('label').addClass('form-check-label').attrs({ for: 'invalidCheck' }).add('Agree to terms and conditions').end().
+            start('div').addClass('invalid-feedback').add('You must agree before submitting.').end().
+          end().
+        end().
+        start('button').addClass('btn').addClass('btn-primary').attrs({ type: 'submit' }).add('Submit form').on('click', this.MyEvent).end().
+        end().
+          
+        start('label').addClass('custom-control-label').attrs({ for: 'customCheck1' }).add('Check this custom checkbox').
+          start().add(this.CHECK_BOX_TEST).addClass(this.checkBoxTest$.map(function(checkBoxTest) {
+            console.log(checkBoxTest);
+            //return checkBoxTest ? self.start('div').addClass('valid-feedback').add().end() : self.start('div').addClass('invalid-feedback').end()
+            //TODO to test
+            })).
+          end().
+        end().
+        start('div').addClass('invalid-feedback').add('Example invalid feedback text').end();
+    }
+  ],
+  listeners: [
+    function MyEvent() {
+      var forms = document.getElementsByClassName('needs-validation');

```

Comment:
This is a very SLOPy (SeLector Oriented Programming) style rather than MVC, which would be much cleaner. myEvent should be an action rather than a listener and its isEnabled button should check if the object is valid.